tests/src/python/test_qgsserver.py

```diff
@@ -117,9 +118,17 @@ def setUp(self):
         self.fontFamily = QgsFontUtils.standardTestFontFamily()
         QgsFontUtils.loadStandardTestFonts(['All'])
 
-        self.testdata_path = unitTestDataPath('qgis_server') + '/'
+        self.temporary_dir = tempfile.TemporaryDirectory()
+        self.temporary_path = self.temporary_dir.name
+
+        # Copy all testdata to the temporary directory
+        copytree(unitTestDataPath('qgis_server'), os.path.join(self.temporary_path, 'qgis_server'))
+        copytree(unitTestDataPath('qgis_server_accesscontrol'), os.path.join(self.temporary_path, 'qgis_server_accesscontrol'))
```

Comment:
Does the test really need to write to these files ?
I'm happy to see commits allowing us to enable CI checking for no source-tree modification, but I also wonder if it's overkill to make a full tree copy (for each test) if the usage pattern is read-only...